node/src/reactor/joiner.rs

```diff
@@ -926,6 +867,153 @@ impl reactor::Reactor for Reactor {
+                    }
+
+                    NetResponse::BlockAndMetadataByHeight(ref serialized_item) =>
+                        match fetcher::Event::<BlockWithMetadata>::from_get_response_serialized_item(
+                            sender,
+                            serialized_item,
+                        ) {
+                            Some(fetcher_event) => self.dispatch_event(
+                                effect_builder,
+                                rng,
+                                JoinerEvent::BlockByHeightFetcher(fetcher_event),
+                            ),
+                            None => {
+                                info!(
+                                    "{} sent us a block with metadata we couldn't parse!
+                Banning",
+                                    sender
+                                );
+                                effect_builder
+                                    .announce_disconnect_from_peer(sender)
+                                    .ignore()
+                            }
+                        }
+
+                    NetResponse::BlockHeaderByHash(ref serialized_item) =>
+                         match fetcher::Event::<BlockHeader>::from_get_response_serialized_item(
+                            sender,
+                            serialized_item,
+                        ) {
+                            Some(fetcher_event) => self.dispatch_event(
+                                effect_builder,
+                                rng,
+                                JoinerEvent::BlockHeaderFetcher(fetcher_event),
+                            ),
+                            None => {
+                                info!(
+                                    "{} sent us a block header we couldn't parse! Banning",
+                                    sender
+                                );
+                                effect_builder
+                                    .announce_disconnect_from_peer(sender)
+                                    .ignore()
+                            }
+                        }
+
+                    NetResponse::BlockHeaderAndFinalitySignaturesByHeight(ref serialized_item) => {
+                       match
+                fetcher::Event::<BlockHeaderWithMetadata>::from_get_response_serialized_item(
+                            sender,
+                        serialized_item,
+                        ) {
+                            Some(fetcher_event) => {
+                                self.dispatch_event(effect_builder, rng,
+                JoinerEvent::BlockHeaderByHeightFetcher(fetcher_event))             } ,
+                            None => {
+                                  info!("{} sent us a block header with finality signatures we
+                couldn't parse! Banning", sender);
+                effect_builder.announce_disconnect_from_peer(sender).ignore()
+                }         }
```

Comment:
That could use some manual formatting… but I can understand why rustfmt didn't feel like doing it. :sweat_smile:

Reply:
Sometimes it just gives up, it seems. :man_shrugging:

Reply:
Did the best I could...